File: Data Structures & Algorithms/longest-consecutive-sequence/submission-9.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const set = new Set(nums)
        let res = 0

        for( let i=0; i<nums.length; i++ ){
            if( set.has( nums[i] - 1 ) ) continue
            let greater = nums[i] + 1
            let currentMax = 0
            while( set.has(greater) ){
                currentMax+=1
                greater+=1
            }
            if(currentMax>=res) res=currentMax+1
        }
        return res
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-9.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int visited[128] = {};
        int res = 0, l = 0;
        for(int r = 0; r < s.length(); r++){
            while(visited[s[r]] != 0){
                visited[s[l]]--;
                l++;
            }
            visited[s[r]]++;
            res = max(res, r - l + 1);
        }
        return res;
    }
};
